lib/heex_formatter/html_tree.ex

```diff
@@ -0,0 +1,75 @@
+defmodule HeexFormatter.HtmlTree do
+  @moduledoc false
+
+  @doc """
+  Build an HTML Tree givens tokens from `Tokenizer.tokenize/1`
+  """
+  def build(tokens) do
+    to_tree(tokens, [], [])
+  end
+
+  def to_tree([], buffer, [:tag_close | tree]), do: to_tree([], buffer, tree)
+  def to_tree([], _buffer, tree), do: Enum.reverse(tree)
+
+  def to_tree([{:tag_open, _n, _a, _m} = token | tokens], [], []) do
+    to_tree(tokens, [], [build_tag_block(token)])
+  end
+
+  def to_tree([{:tag_open, _n, _a, _m} = token | tokens], buffer, []) do
```

Comment:
Maybe it's too early to comment since it's in WIP, but just a reminder, can we use var names that give more context?